### PR TITLE
fix: handle bad keyfiles

### LINKED ISSUE #141: Insufficient error message for bad keys
**Describe the bug**

If one of the keys contains some error, for example the json is invalid or the key is not a valid key, GCP Scanner crashes with a very unhelpful error message.

**To Reproduce**
1. Put a few keys in a directory with at least one of them invalid format
2. Run gcp scanner pointing to that directory

**Expected behavior**

Either
1. A warning log telling me exactly which key failed to parse and then continue
2. An error log telling me exactly which key failed to parse and then stop
3. An error log telling me exactly which key failed to parse, then continue parsing (for more potential errors) and then stop without scanning (so that you can handle all errors in bulk instead of fixing one and re-running.

**Current behavior**

GCP Scanner crashes with an error telling me that it failed to parse some key

### PR DESCRIPTION
**Pull request by csemanish12**
closes #141 
- moved the logic of fetching sa_credentials to a method
- handled exception while fetching sa_credentials from keyfiles
- added error log for all the malformed keyfiles

### CODE REVIEW

**Comment by mshudrak** [comment]
Please accept Google CLA before we can proceed with review of this code.

File: src/gcp_scanner/scanner.py [hunk 1]
```diff
@@ -475,6 +477,31 @@ def gke_client_for_credentials(
     credentials=credentials)
 
 
+def get_sa_details_from_key_files(key_path):
```
Comment (mshudrak) [inline src/gcp_scanner/scanner.py:480]: Looks good. Could you please add a test for this function? One that fails and one that succeed.

**Review by mshudrak — CHANGES_REQUESTED** [review]
Thanks for PR. Looks good but we need to add a test for new function. Additionally, tests are failing, PTAL, seems like pylint is complaining about something.

File: src/gcp_scanner/scanner.py (hunk 1 above)
Comment (csemanish12) [reply src/gcp_scanner/scanner.py:480]: done

**Pull request merged**